### PR TITLE
🎉 Source Stripe: add SetupIntents stream

### LINKED ISSUE #16978: Source Stripe: new stream SetupIntents
## Tell us about the problem you're trying to solve
[SetupIntents](https://stripe.com/docs/api/setup_intents)
A SetupIntent guides you through the process of setting up and saving a customer's payment credentials for future payments. For example, you could use a SetupIntent to set up and save your customer's card without immediately collecting a payment. Later, you can use [PaymentIntents](https://stripe.com/docs/api/setup_intents#payment_intents) to drive the payment flow.

[Permission](https://stripe.com/docs/security/permissions) _setup_intent_read, setup_intent_write_ Grants access to [SetupIntents](https://stripe.com/docs/api/setup_intents). This permission also implies the following permission: _mandate_read_

## Describe the solution you’d like
Research pros and cons for implementing this stream.
_Research should contain: what stream should be implemented, permissions and scopes (if needed), paid plan (if needed)._

### PR DESCRIPTION
**Pull request by jj-nilbodee**
## What
Add incremental `SetupIntents` stream to source Stripe
Closes #16978

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

<summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by jhajajaas** [comment]
Unit test results:

Screenshot 2023-03-13 at 15 21 05

**Comment by marcosmarxm** [comment]
Thanks @jhajajaas for the contribution, looks another contribution was merged and there are some conflicts, can you solve them? I'll add this task to my backlog.

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml [hunk 1]
```diff
@@ -31,6 +32,8 @@ acceptance_tests:
             bypass_reason: "This stream can't be seeded in our sandbox account"
           - name: "transfers"
             bypass_reason: "This stream can't be seeded in our sandbox account"
+          - name: "products"
+            bypass_reason: "This stream can't be seeded in our sandbox account"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:36]: Why bypassing this stream?

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/setup_intents.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+    "type": ["null", "object"],
+    "properties": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/setup_intents.json:3]: missing additionalProperties = true.

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-stripe/source_stripe/schemas/setup_intents.json:3]: check others streams in the same source.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml (hunk 1 above)
Comment (jhajajaas) [reply airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:36]: Sorry I wasn't sure with the empty_streams. Do we put streams in here when it _could_ be empty? I mean the credentials I tested has no products so the integration test fails if I don't bypassing it here.

**Comment by jhajajaas** [comment]
That's weird. It failed because of customers stream but I didn't touch that. Could this be because of the latest merge on customers stream?

**Comment by sajarin** [comment]
Hey thanks for the contribution and apologies for the delay! 

I'll be the DRI responsible for reviewing and getting across the finish line. We currently have a backlog of about a dozen PRs, but just wanted to let you know that this PR has been added to the queue and I'll be trying my best to get to this as soon as possible. 

Thanks for the contribution and for being patient :)

**Review by marcosmarxm — APPROVED** [review]
Thanks @jhajajaas the tests are working now. We're going to publish and merge the contirbution during the week.

**Review by bazarnov — APPROVED** [review]
Looks good, let's make sure the CAT passes here, otherwise LGTM.

**Pull request merged**